File: api/src/main/java/put/poznan/sport/entity/TrainingSession.java
package put.poznan.sport.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import put.poznan.sport.entity.sportFacility.SportFacility;

import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TrainingSession {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private Integer coachId;
    private Integer sportFacilityId;
    private String name;
    private String startHour;
    private Integer duration;
    private Integer isWeekly;
    private Date trainingDate;
    private Integer capacity;
    private Integer freeBooked;

    @ManyToOne
    @JoinColumn(name = "coachId", insertable = false, updatable = false)
    @JsonIgnore
    private Coach coach;

    @ManyToOne
    @JoinColumn(name = "sportFacilityId", insertable = false, updatable = false)
    @JsonIgnore
    private SportFacility sportFacility;

    @OneToMany(mappedBy = "trainingSession", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonIgnore
    private List<TrainingSessionParticipant> trainingSessionParticipants;
}

File: mobile_app/lib/models/training_session.dart
import 'package:sport_plus/models/coach.dart';
import 'package:sport_plus/models/rating.dart';

class TrainingSession {
  final int id;
  final Coach coach;
  final String name;
  final int startHour;
  final int duration;
  final bool isWeekly;
  final DateTime trainingDate;
  final int capacity;
  final bool freeBooked;
  final Rating rating;

  TrainingSession(
      {required this.id,
      required this.coach,
      required this.name,
      required this.startHour,
      required this.duration,
      required this.isWeekly,
      required this.trainingDate,
      required this.capacity,
      required this.freeBooked,
      required this.rating});
}
